Number_types/test/Number_types/to_interval_test_boost.cpp

```diff
@@ -0,0 +1,867 @@
+
+  // std::cout << std::endl;
+  // std::cout << boost::typeindex::type_id<NT>() << std::endl;
+  // std::cout << std::endl;
+  // std::cout << boost::typeindex::type_id<typename Traits::Type>() << std::endl;
+  // std::cout << std::endl;
+
+  NT x;
+  double i, s;
+
+  std::cout << std::endl;
+  std::cout << "- T testing tight interval for integers ..." << std::endl;
+  std::cout << std::endl;
+
+  #ifdef TESTCASE_INT_0
+
+  std::cout << "=============" << std::endl;
+  std::cout << "CASE0 RESULT:" << std::endl;
+  std::cout << "=============" << std::endl;
+
+  x = NT("0");
+  std::tie(i, s) = Interval()(x);
+
+  std::cout << std::endl;
+  std::cout << "inf: " << i << std::endl;
+  std::cout << "ref: 0" << std::endl;
+  std::cout << "sup: " << s << std::endl;
+  std::cout << "ref: 0" << std::endl;
+  std::cout << std::endl;
+
+  assert(i == 0.0);
+  assert(s == 0.0);
+
+  #endif
+
+  #ifdef TESTCASE_INT_1
+
+  std::cout << "=============" << std::endl;
+  std::cout << "CASE1 RESULT:" << std::endl;
+  std::cout << "=============" << std::endl;
+
+  x = NT("5");
+  std::tie(i, s) = Interval()(x);
+
+  std::cout << std::endl;
+  std::cout << "inf: " << i << std::endl;
+  std::cout << "ref: 5" << std::endl;
+  std::cout << "sup: " << s << std::endl;
+  std::cout << "ref: 5" << std::endl;
+  std::cout << std::endl;
+
+  assert(i == 5.0);
+  assert(s == 5.0);
+
+  #endif
+
+  #ifdef TESTCASE_INT_2
+
+  std::cout << "=============" << std::endl;
+  std::cout << "CASE2 RESULT:" << std::endl;
+  std::cout << "=============" << std::endl;
+
+  x = NT(-12);
+  std::tie(i, s) = Interval()(x);
+
+  std::cout << std::endl;
+  std::cout << "inf: " << i << std::endl;
+  std::cout << "ref: -12" << std::endl;
+  std::cout << "sup: " << s << std::endl;
+  std::cout << "ref: -12" << std::endl;
+  std::cout << std::endl;
+
+  assert(i == -12.0);
+  assert(s == -12.0);
+
+  #endif
+
+  #ifdef TESTCASE_INT_3
+
+  std::cout << "=============" << std::endl;
+  std::cout << "CASE3 RESULT:" << std::endl;
+  std::cout << "=============" << std::endl;
+
+  x = NT("772537196160711547532081795586792063331305895970601529435744397743492241616327030886637827664482971614281724796166908515292029740442872965475211471498392497954317530347232852540146110053764627070672243390766540271554856759037331142360111552286202392826786995364211101723592791550906796165626083442695020580821188398298798456115881346136681033873");
+  std::tie(i, s) = Interval()(x);
+
+  std::cout << std::endl;
+  std::cout << "inf: " << i << std::endl;
+  std::cout << "ref: 1.7976931348623157081e+308" << std::endl;
+  std::cout << "sup: " << s << std::endl;
+  std::cout << "ref: inf" << std::endl;
+  std::cout << std::endl;
+
+  assert(i == std::numeric_limits<double>::max());
+  assert(s == std::numeric_limits<double>::infinity());
+
+  #endif
+
+  std::cout << "---SUCCESS ALL---" << std::endl;
+  std::cout << std::endl;
+}
+
+void test_shift_positive() {
+  {
+    double d = (1L << 53) - 1;
```

Comment:
```suggestion
    double d = (1LL << 53) - 1;
```
(same for the other tests below)
On platforms with 32-bit long it won't fit otherwise.